cmake/bintools/oneApi/target.cmake

```
@@ -1,12 +1,18 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(DEFINED TOOLCHAIN_HOME)
-  set(find_program_clang_args PATHS ${TOOLCHAIN_HOME} ${ONEAPI_PYTHON_PATH} NO_DEFAULT_PATH)
+  set(find_program_clang_args PATHS ${TOOLCHAIN_HOME} ${ONEAPI_LLVM_BIN_PATH} NO_DEFAULT_PATH)
   set(find_program_binutils_args PATHS ${TOOLCHAIN_HOME} )
 endif()
 
+find_package(oneApi 2023.0.0)
```

you should probably do:
```suggestion
find_package(oneApi 2023.0.0 REQUIRED)
```

in which case there shouldn't be a need to do the `set(CMAKE_C_COMPILER CMAKE_C_COMPILER-NOTFOUND ...)` or similar code later.
Code which can the be removed.